File: 5 Kyu/5 kyu - Is my friend cheating?.c
#include <stdlib.h>

typedef struct _pair { long long first, snd; } Pair;

Pair *mkpair (long long a, long long b) {
    Pair *p = malloc (sizeof(Pair));
    p->first = a, p->snd = b;
    return p;
}
Pair** removNb(long long n, int* length) {
    long long cnt = n * (n + 1) / 2, a = n, b, sum;
    Pair **seq = malloc (5 * sizeof(Pair));
    int index = 0;
    
    while (a-->2) {
        b = (cnt - a) / (a + 1), sum = cnt - a - b;
        if (b < n && a * b == sum) seq[index++] = mkpair (b,a);
    }
  
    *length = index; 
    return seq;
}

File: 5 Kyu/5 kyu - Is my friend cheating?.rs
fn remove_nb (m: i32) -> Vec<(i32, i32)> {
    let n:i64 = m as i64;
    let cnt:i64 = n * (n + 1) / 2;
    let mut seq = Vec::<(i32,i32)>::new();
    
    for a in 2..n {
        
        let b = (cnt - a) / (a + 1);
        let sum = cnt - a - b;
        
        if b < n && a * b == sum {
            seq.push ((a as i32,b as i32));
        }
    }
    
    return seq;
}
